package/base-files/files/lib/functions/system.sh

```diff
@@ -42,48 +49,67 @@ get_mac_label_json() {
 		json_select ..
 	fi
 
-	echo $macaddr
+	echo -n "$macaddr"
 }
 
 get_mac_label() {
-	local macaddr=$(get_mac_label_dt)
+	local macaddr
 
-	[ -n "$macaddr" ] || macaddr=$(get_mac_label_json)
+	[ -n "$(get_mac_label_dt)" ] &&
```

Comment:
This invokes `get_mac_label_dt()` twice in the worst case. The result should be cached in an intermediate variable.